op-node/flags: Add easy way to deprecate flags

trianglesphere opened the pull request:
**Description**

This adds a deprecated flags section to warn users that the flag will be removed in a future version of the software. I had to modify the P2P Flags structure to make it work nicely.

I also added a test to ensure that deprecated flags are hidden & that all flags have at least one env var.

trianglesphere commented on op-service/util.go:
```diff
@@ -62,6 +62,14 @@ func validateEnvVars(prefix string, providedEnvVars []string, definedEnvVars map
 	return out
 }
 
+func WarnOnDeprecatedFlags(ctx *cli.Context, deprecatedFlags []cli.Flag, log log.Logger) {
```
```suggestion
// WarnOnDeprecatedFlags iterates through the provided deprecatedFlags and logs a warning for each that is set.
func WarnOnDeprecatedFlags(ctx *cli.Context, deprecatedFlags []cli.Flag, log log.Logger) {
```

trianglesphere committed:
Update op-service/util.go

sebastianst commented on op-node/flags/p2p_flags.go:
```diff
@@ -55,6 +55,30 @@ var (
 	SyncReqRespName        = "p2p.sync.req-resp"
 )
 
+func DeprecatedP2PFlags(envPrefix string) []cli.Flag {
```
Why does this func need to be public?

trianglesphere replied:
oh, p2pdeprecated doesn't. Deprecated does

sebastianst reviewed (commented):
LGTM, one q

trianglesphere committed:
op-node/flags: Add easy way to deprecate flags

This adds a deprecated flags section to warn users that the flag will be removed
in a future version of the software. I had to modify the P2P Flags structure to
make it work nicely.

I also added a test to ensure that deprecated flags are hidden & that all flags
have at least one env var.
